awx/main/dispatch/reaper.py

```diff
@@ -14,6 +15,17 @@ def reap_job(j, status):
     if UnifiedJob.objects.get(id=j.id).status not in ('running', 'waiting'):
         # just in case, don't reap jobs that aren't running
         return
+    if j.instance_group.is_container_group:
+        status, detail = receptor_work_status(j.work_unit_id)
+        if RECEPTOR_AWX_STATE_MAP[status] != j.status:
+            # Sometimes the job has exited already, but we had not had chance to update the
+            # job status with one of these terminal states.
+            # Update the status and return early.
+            # Otherwise the job truly is gone but still marked as running, in which case it should
+            # rightfully be reaped
```

Comment:
This workaround fixed some jobs that were getting reaped when they actually were healthy. But I do wonder if this is necessary because I'm missing something earlier in the lifecycle of the job.